### ISSUE TITLE
XLSX tab naming can cause IllegalArgumentException

### ISSUE THREAD

**Issue by SteveSchafer-Innovent** [eclipse/birt#641]
The spudsoft emitter uses the table of contents entries for tab naming.  POI and/or Excel has some restrictions on what characters can comprise a tab name, and this can lead errors that can be data-dependent.  Additionally there is a size limit on the name but the emitter code doesn't take that into account when searching for duplicate names, which can also lead to errors.  The latter issue is definitely a bug.  For the first issue I think it would be acceptable for the emitter to edit the names before setting them, although this might be a matter of opinion.   Please leave comments as to whether you agree.

Related bugzilla: https://bugs.eclipse.org/bugs/show_bug.cgi?id=441382

I can implement these fixes but I don't know of yaytay (spudsoft author) is still involved.

**Comment by hvbargen** [comment]
Steve, in my fork at https://github.com/hvbtup/birt I updated the XLSX emitter to the latest version of Jim's (yaytay) code when it was still available. The repository was at bitbucket.org but is no longer available. 
https://github.com/Yaytay is very likely the same person - I don't know if he's still interested in BIRT.
I don't know if this might help with this issue, I just wanted to let you know.

I stumbled  over this issue myself once, and I think it's OK if the emitter replaces invalid characters with a replacement characters (for example, "_") or removes them, and if the emitter truncates the name.
Anyway, if the emitter has to change the name, this should be logged.

This is much better than a lengthy error stack trace.
